k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/KustomizeResourceHandler.kt

```diff
@@ -1,66 +1,134 @@
+            "kind" to artifact.kind,
+            "name" to "${artifact.name}-${environment.name}",
+            "namespace" to artifact.namespace
+        )
+        spec?.set("sourceRef", sourceRef)
+
+        val repoUrl = artifactFromKeelRepository?.gitMetadata?.repo?.link
+            ?: throw InvalidArtifact("artifact version $version does not have repository URL in artifact metadata")
+        val gitRepoManifest = generateGitRepoManifest(artifact, repoUrl, version, environment.name)
+        resource.spec.template = toK8sList(
+            gitRepoManifest,
+            resource.spec.template,
+            generateCorrelationId(resource)
+        )
 
         // sending it to the super class for common labels and annotations to be added
         return super.toResolvedType(resource)
     }
 
-    override suspend fun current(resource: Resource<KustomizeResourceSpec>): K8sObjectManifest? =
-        super.current(resource)?.let {
-            val lastAppliedConfig = (it.metadata[ANNOTATIONS] as Map<String, String>)[K8S_LAST_APPLIED_CONFIG] as String
-            return cleanup(jacksonObjectMapper().readValue(lastAppliedConfig))
-        }
+    override suspend fun current(resource: Resource<KustomizeResourceSpec>): K8sObjectManifest? {
+        val deployed = getK8sResource(resource)
+        // Check health of resources returned by clouddriver
+        notifyHealthAndArtifactDeployment(deployed, resource)
+        return deployed
+    }
 
+    @Suppress("UNCHECKED_CAST")
     override suspend fun getK8sResource(r: Resource<KustomizeResourceSpec>): K8sObjectManifest? =
-    // defer to GenericK8sResourceHandler to get the resource
-        // from the k8s cluster
-        cloudDriverK8sService.getK8sResource(r)?.let {
-            it.manifest.to<K8sObjectManifest>()
+        coroutineScope {
+            val environment = repository.environmentFor(r.id)
+            val (artifact, _) = getArtifactAndConfig(r)
+            super.getFluxK8sResources(r, artifact, generateCorrelationId(r), environment.name)
         }
 
     override suspend fun actuationInProgress(resource: Resource<KustomizeResourceSpec>): Boolean =
         resource
             .spec.template.let {
-                orcaService.getCorrelatedExecutions(it.name()).isNotEmpty()
+                log.debug("Checking if actuation is in progress")
+                orcaService.getCorrelatedExecutions(generateCorrelationId(resource)).isNotEmpty()
             }
+
+    // Flux only supports Git as its SourceRepository
+    private fun getArtifactAndConfig(resource: Resource<KustomizeResourceSpec>): Pair<GitRepoArtifact, DeliveryConfig> {
+        val deliveryConfig = repository.deliveryConfigFor(resource.id)
+        resource.spec.artifactRef.let { tagRef ->
+            val artifact = deliveryConfig.artifacts.find {
+                log.trace("checking $it")
+                it.reference == tagRef && it.type.toUpperCase() == FluxSupportedSourceType.GIT.name
```

Comment:
do we need this check here? looks like this will have to be dealt with in artifact handler and then we should be able to have only one function to verify it both for HELM and Kustomize.

Reply:
Yeah this was one of functions couldn't be put into generic handler without another abstraction on top of it. I'll move it